packages/docs/src/components/unobtrusive-consent-control-ui.js

```diff
@@ -0,0 +1,197 @@
+  const initial = useMemo(() => {
+    const initialState = {}
+    for (const integration of integrations) {
+      initialState[integration.id] = initialValues.enabled.includes(
+        integration.id
+      )
+    }
+
+    return initialState
+  }, [integrations, initialValues])
+
+  // Check if component was mounted for SSR
+  const [isMounted, setIsMounted] = useState(false)
+  useEffect(() => setIsMounted(true), [setIsMounted])
+
+  // Introduction animation
+  useEffect(() => {
+    if (!isMounted || !needsIntroduction) {
+      return
+    }
+    setShowIntroduction(true)
+    setNeedsIntroduction(false)
+
+    // Wait for no user interaction
+    // @todo actually check it
+    window.setTimeout(() => {
+      setSlideUp(true)
+
+      // Keep it visible to the user
+      window.setTimeout(() => {
+        setSlideUp(false)
+
+        // Wait with swapping contentn till slide down is finished
+        window.setTimeout(() => {
+          setShowIntroduction(false)
+        }, introductionTransitionDuration)
+      }, introductionVisibleDuration)
+    }, introductionNoActionDelay)
+  }, [
+    isMounted,
+    setShowIntroduction,
+    needsIntroduction,
+    introductionNoActionDelay,
+    introductionVisibleDuration,
+    introductionTransitionDuration,
+  ])
+
+  // Do not render the interface on SSR.
+  if (!isMounted) {
+    return null
+  }
+
+  return (
+    <div
+      className={clsx(
+        styles.wrapper,
+        styles.pane,
+        styles.slide,
+        slideUp && styles.slideUp
+      )}
+      style={{
+        transitionDuration: `${introductionTransitionDuration}ms`,
+      }}
+      id="consent-control-ui"
+    >
+      <button
+        className={clsx(styles.toggleButton)}
+        title={`Toggle website settings visibility`}
+        onClick={toggleControlForm}
+      >
+        <div className={clsx(styles.pane, styles.toggleButtonContent)}>
+          <ToggleIcon className={clsx(slideUp && styles.inverted)} />
+        </div>
+      </button>
+      {showIntroduction ? (
+        <div className={clsx(styles.introduction)}>
+          Some features got disabled in respect of your privacy.
+          <br /> Click here to learn more!
+        </div>
+      ) : (
+        <Form
+          onSubmit={onSubmitCb}
+          initialValues={initial}
+          render={({ handleSubmit }) => (
+            <form onSubmit={handleSubmit}>
+              <div className={clsx(styles.content)}>
+                <h2>Website Settings</h2>
+                <p>
+                  Some features are disabled by default to protect your privacy:
+                </p>
+                {integrations.map(
+                  ({
+                    id,
+                    Icon,
+                    title,
+                    contrastColor,
+                    color,
+                    description,
+                    privacyPolicyUrl,
+                  }) => (
+                    <div className={clsx(styles.formControl)} key={id}>
```

Comment:
Can we move the integration fields to their own component or would that make things significantly more complex?

Reply:
Should be improved in new state, but we still might want to move some stuff into their own components.